nettacker/core/graph.py

```diff
@@ -119,6 +121,106 @@ def create_compare_text_table(results):
     return table.draw() + "\n\n"
 
 
+def create_dd_specific_json(all_scan_logs):
+    severity_mapping = {1: "Info", 2: "Low", 3: "Medium", 4: "High", 5: "Critical"}
+
+    findings = []
+
+    module_path = Config.path.modules_dir
+
+    for log in all_scan_logs:
+        module_name = log["module_name"].strip()
+        date = log["date"].split(" ")[0].strip()
+        port = log.get("port", "").__str__().strip()
+        impact = log.get("event", "").strip()
+        severity_justification = log.get("json_event", "").strip()
+        service = log.get("target", "").strip()
+        unique_id = log.get("scan_id", uuid.uuid4().hex)
+
+        module_file = f"{str(module_path).strip()}/{module_name.split('_')[-1].strip()}/{'_'.join(module_name.split('_')[0:-1])}.yaml"
+
+        try:
+            with open(module_file) as fp:
+                data = yaml.safe_load(fp)
+                severity_raw = data["info"].get("severity", 0)
+                description = data["info"].get("description", "")
+        except Exception:
+            severity_raw = 1  # Default to Info
+            description = "No description available."
+
+        if severity_raw >= 9:
+            severity = severity_mapping[5]
+        elif severity_raw >= 7:
+            severity = severity_mapping[4]
+        elif severity_raw >= 4:
+            severity = severity_mapping[3]
+        elif severity_raw > 0:
+            severity = severity_mapping[2]
+        else:
+            severity = severity_mapping[1]
+
+        finding = {
```

Comment:
No need to declare `finding` as you use it only once -- just append the dict directly.